explainer-server/app/controllers/ApiController.scala

```diff
@@ -76,6 +76,50 @@ class ExplainerApiImpl(
     explainerToPublish.map(CsAtom.atomToCsAtom)
   }
 
+  override def addTagToExplainer(explainerId: String, tagId: String): Future[CsAtom] = {
+    for {
+      csatom <- explainerDB.load(explainerId).map(CsAtom.atomToCsAtom)
+    }yield{
+      val csexplaineratom: CsExplainerAtom = csatom.data
+      val tags: Option[List[String]] = csexplaineratom.tags
+      val newtags: Option[List[String]] = tags match {
+        case None => Some(List(tagId))
+        case Some(list) => Some( (tagId :: list).distinct.sorted )
+      }
+      val newCsExplainerAtom: CsExplainerAtom = CsExplainerAtom(csexplaineratom.title, csexplaineratom.body, csexplaineratom.displayType, newtags)
+      val newCsAtom: CsAtom = CsAtom(csatom.id, newCsExplainerAtom, csatom.contentChangeDetails)
+      explainerDB.store(CsAtom.csAtomToAtom(newCsAtom))
+      explainerStore.updateLastModified(explainerId, user)
+    }
+    load(explainerId)
+  }
+
+  override def removeTagFromExplainer(explainerId: String, tagId: String): Future[CsAtom] = {
+    for {
+      csatom <- explainerDB.load(explainerId).map(CsAtom.atomToCsAtom)
+    }yield{
+      val csexplaineratom: CsExplainerAtom = csatom.data
+      val tags: Option[List[String]] = csexplaineratom.tags
+      val newtags: Option[List[String]] = tags match {
+        case None => None
```

Comment:
I indeed cannot use map here because I still need to return `None` even if the tags wasn't `None`. Now using `.isEmpty` and shorted code

```
  override def removeTagFromExplainer(explainerId: String, tagId: String): Future[CsAtom] = {
    explainerDB.load(explainerId).map(CsAtom.atomToCsAtom).map{ csatom =>
      val newCsAtom: CsAtom = csatom.copy(
        data = csatom.data.copy(
          tags = csatom.data.tags.flatMap{ list =>
            val newlist = (list diff List(tagId)).sorted
            if(newlist.isEmpty){ None }else{ Some(newlist) }
          }
        )
      )
      explainerDB.store(CsAtom.csAtomToAtom(newCsAtom))
      explainerStore.updateLastModified(explainerId, user)
    }
    load(explainerId)
  }
```

Momentarily ignoring Phil's suggestion on the right moment to use the CS to nonCS mapping